riscv/src/small_field/code_gen.rs

```diff
@@ -0,0 +1,1822 @@
+use std::vec;
+
+use itertools::Itertools;
+use powdr_isa_utils::SingleDataValue;
+use powdr_isa_utils::{escape_label, quote};
+
+use powdr_number::KnownField;
+
+use crate::continuations::bootloader::{bootloader_and_shutdown_routine, bootloader_preamble};
+
+use crate::code_gen::{
+    InstructionArgs, MemEntry, Register, RiscVProgram, SourceFileInfo, Statement,
+};
+use crate::CompilerOptions;
+
+use crate::small_field::runtime::Runtime;
+
+/// Translates a RISC-V program to POWDR ASM
+/// with constraints that work for a field >16 bits.
```

Comment:
is the `>16` part correct? i guess it is for the asm machine right? only the submachines are more field specific?

Reply:
The asm machines are also specific to field size, though some require 16, 17, 18, or more bits. Poseidon is field specific. The RISCV asm machine here requires 16 bits for the limbs, but in some places it does require more. I'll update this comment.